src/MainWindow/main_window.cpp

```diff
@@ -793,6 +789,37 @@ void MainWindow::pinAssignmentActionTriggered() {
   saveToolBar->setHidden(!pinAssignmentAction->isChecked());
 }
 
+void MainWindow::ipConfiguratorActionTriggered() {
+  if (ipConfiguratorAction->isChecked()) {
+    IpConfiguratorCreator creator;
+    // Available IPs DockWidget
+    m_availableIpsgDockWidget = PrepareTab(tr("IPs"), "availableIpsWidget",
+                                           creator.GetAvailableIpsWidget(),
+                                           nullptr, Qt::RightDockWidgetArea);
+
+    // Get the actual IpCatalogTree
+    auto ipsWidgets = m_availableIpsgDockWidget->findChildren<IpCatalogTree*>();
+    if (ipsWidgets.count() > 0) {
+      m_ipCatalogTree = ipsWidgets[0];
+
+      // Update the IP Config widget when the Available IPs selection changes
+      QObject::connect(m_ipCatalogTree, &IpCatalogTree::itemSelectionChanged,
```

Comment:
Are u sure this connection won't be duplicated after call this slot few times?

Reply:
Yeah, m_ipCatalogTree gets deleted (by cleanUpDockWidgets()) each time the view action is toggled to off so the connect will be removed before a new one is created next time the view is selected. To be sure, I put in a print statement in the callback and then opened/closed the view a few times and confirmed only 1 print was done when the handler fired.